nncf/common/pruning/operations.py

```diff
@@ -244,6 +319,21 @@ class ReshapePruningOp(BasePruningOp):
     def _is_flatten(node: NNCFNode) -> bool:
         return len(node.layer_attributes.output_shape) == 2
 
+    @staticmethod
+    def _is_not_mixing_dim(node: NNCFNode, graph) -> bool:
+        input_shape = node.layer_attributes.input_shape
+
+        output_edges = graph.get_output_edges(node)
+        if not output_edges:
+            return True
+        output_shape = node.layer_attributes.output_shape
+
+        if len(input_shape)==len(output_shape) and set(input_shape)==set(output_shape):
+            input_shape_idx = np.argsort(input_shape)
+            output_shape_idx = np.argsort(output_shape)
+            return np.allclose(input_shape_idx, output_shape_idx)
+        return True
```

Comment:
This is equivalent to this:
```suggestion
        if len(input_shape) != len(output_shape):
            return True
        return input_shape == output_shape
```

Reply:
I don't understand why we need this set comparison. 
Do we really want to propagate through such things?
(a, c, b) -> (a * b, c, 1) and etc?

Reply:
I tried to cover (b,a) -> (a, b) in here. The case of (a, c, b) -> (a*b, c, 1) is not covered in here.